global/scripts/MSMSulc.sh

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
+set -eu
+
+pipedirguessed=0
+if [[ ""$HCPPIPEDIR:-"" == "" ]]
+then
+    pipedirguessed=1
+    #fix this if the script is more than one level below HCPPIPEDIR
+    export HCPPIPEDIR="$(dirname -- "$0")/.."
+fi
+
+source "$HCPPIPEDIR/global/scripts/newopts.shlib" "$@"
+source "$HCPPIPEDIR/global/scripts/debug.shlib" "$@"
+
+#description of script/command
+opts_SetScriptDescription "Run MSMSulc registration and save distortion outputs"
+
+opts_AddMandatory '--subject-dir' 'SubjectDir' 'path' "folder containing all subjects"
+opts_AddMandatory '--subject' 'Subject' 'subject ID' "subject-id"
+opts_AddOptional '--regname' 'RegName' 'my reg' "set a new registration name, default MSMSulc" 'MSMSulc'
+opts_AddOptional '--msm-conf' 'ConfFile' 'conf file' "provide the name of the configuration file, default MSMSulcStrainFinalconf" "$MSMCONFIGDIR"/MSMSulcStrainFinalconf
+opts_AddOptional '--refmesh' 'RefMesh' 'ref mesh' "provide alternate standard sphere, default 164k_fs_LR, use .HEMISPHERE. instead of .L. or .R."
+opts_AddOptional '--refdata' 'RefData' 'ref data' "provide alternate reference data, use .HEMISPHERE. instead of .L. or .R."
+
+opts_ParseArguments "$@"
+
+if ((pipedirguessed))
+then
+    log_Err_Abort "HCPPIPEDIR is not set, you must first source your edited copy of Examples/Scripts/SetUpHCPPipeline.sh"
+fi
+
+#display the parsed/default values
+opts_ShowValues
+
+#set paths
+SurfaceTemplateFolder="$HCPPIPEDIR"/global/templates/standard_mesh_atlases
+NonlinearFolder="$SubjectDir"/"$Subject"/MNINonLinear
+NativeFolder="$NonlinearFolder"/Native
+
+#if user provided --refmesh but not --refdata, scream
+if [[ "$RefMesh" != "" && "$RefData" == "" ]]
+then
+	log_Err_Abort "Non-default standard sphere provided, but no non-default reference data"
+fi
+
+if [[ "$RefMesh" == "" ]]
+then
+	RefMesh="$SurfaceTemplateFolder"/fsaverage.HEMISPHERE_LR.spherical_std.164k_fs_LR.surf.gii
+fi
+
+if [[ "$RefData" == "" ]]
+then
+	RefData="$SurfaceTemplateFolder"/HEMISPHERE.refsulc.164k_fs_LR.shape.gii
+fi
+
+#do the same with the conf file
+
+#Make MSMSulc Directory
+mkdir -p "$NativeFolder"/"$RegName"
+
+#Loop through left and right hemispheres
+for Hemisphere in L R ; do
+
+	if [[ "$Hemisphere" == "L" ]] ; then
+		Structure="CORTEX_LEFT"
+	elif [[ "$Hemisphere" == "R" ]] ; then
+		Structure="CORTEX_RIGHT"
+	fi
+	
+	#Calculate Affine Transform and Apply
+	if [ ! -e "$NativeFolder"/"$Subject"."$Hemisphere".sphere.rot.native.surf.gii ] ; then
```

Comment:
I would argue that we shouldn't be doing this test in the pipelines version.  The existing PostFS script creates sphere.rot unconditionally:

https://github.com/Washington-University/HCPpipelines/blob/53e253ccd8105fd91ea202807407ad62beb2417a/PostFreeSurfer/scripts/FreeSurfer2CaretConvertAndRegisterNonlinear.sh#L421